File: windows/QMK Toolbox/Usb/Bootloader/KiibohdDfuDevice.cs
﻿using System.IO;
using System.Threading.Tasks;

namespace QMK_Toolbox.Usb.Bootloader
{
    class KiibohdDfuDevice : BootloaderDevice
    {
        public KiibohdDfuDevice(UsbDevice d) : base(d)
        {
            Type = BootloaderType.KiibohdDfu;
            Name = "Kiibohd DFU";
            PreferredDriver = "WinUSB";
            IsResettable = true;
        }

        public async override Task Flash(string mcu, string file)
        {
            if (Path.GetExtension(file)?.ToLower() == ".bin")
            {
                await RunProcessAsync("dfu-util.exe", $"-a 0 -d 1C11:B007 -D \"{file}\"");
            }
            else
            {
                PrintMessage("Only firmware files in .bin format can be flashed with dfu-util!", MessageType.Error);
            }
        }

        public async override Task Reset(string mcu) => await RunProcessAsync("dfu-util.exe", "-a 0 -d 1C11:B007 -e");
    }
}

File: linux/src/qmk_toolbox/bootloader/kiibohd_dfu_device.py
from .bootloader_device import BootloaderDevice
from .bootloader_type import BootloaderType


class KiibohdDfuDevice(BootloaderDevice):
    def __init__(self, usb_device):
        super().__init__(usb_device)
        self.bootloader_type = BootloaderType.KIIBOHD_DFU
        self.name = "Kiibohd DFU"
    
    async def flash(self, mcu: str, file_path: str) -> int:
        args = ["-d", f"{self.vendor_id:04x}:{self.product_id:04x}", "-D", file_path]
        return await self._run_process("dfu-util", args)
